system/jlib/jbuff.cpp

```diff
@@ -64,9 +64,16 @@
 #define     CHECKREADPOS(len)  
 #endif
 
-
 //-----------------------------------------------------------------------
 
+static inline size32_t checkMemoryBufferOverflow(size32_t curLen, size32_t inc)
+{
+    size_t newLen = (size_t)curLen + (size_t)inc;
+    if (newLen >= MEMBUFFER_MAXLEN)
```

Comment:
trivial: technically it's okay to be == to MEMBUFFER_MAXLEN, a problem if > only.